src/app/system/external-services/payment-provider/add-payment-provider/add-payment-provider.component.html

```diff
@@ -0,0 +1,156 @@
+          <div fxFlex="40%" fxLayout="column">
+            <mat-form-field>
+              <mat-label>Account creation endpoint</mat-label>
+              <input matInput required formControlName="account_creation_endpoint" />
+              <mat-error *ngIf="addPaymentProviderForm.controls.account_creation_endpoint.hasError('required')">
+                Account creation endpoint is <strong>required</strong>
+              </mat-error>
+            </mat-form-field>
+          </div>
+        </div>
+
+        <div fxLayout="row">
+          <div fxFlex="40%" fxLayout="column">
+            <mat-form-field>
+              <mat-label>Authentication endpoint</mat-label>
+              <input matInput required formControlName="authentication_endpoint" />
+              <mat-error *ngIf="addPaymentProviderForm.controls.authentication_endpoint.hasError('required')">
+                Authentication endpoint is <strong>required</strong>
+              </mat-error>
+            </mat-form-field>
+          </div>
+        </div>
+
+        <div fxLayout="row">
+          <div fxFlex="40%" fxLayout="column">
+            <mat-form-field>
+              <mat-label>Authentication type</mat-label>
+              <mat-select required formControlName="authentication_type">
+                <mat-option *ngFor="let authenticationType of authenticationTypeOptions" [value]="authenticationType.id">
+                  {{ authenticationType.name }}
+                </mat-option>
+              </mat-select>
+              <mat-error *ngIf="addPaymentProviderForm.controls.authentication_type.hasError('required')">
+                Authentication type is <strong>required</strong>
+              </mat-error>
+            </mat-form-field>
+          </div>
+        </div>
+
+        <div fxLayout="row">
+          <div fxFlex="40%" fxLayout="column">
+            <mat-form-field>
+              <mat-label>Username</mat-label>
+              <input matInput required formControlName="username" />
+              <mat-error *ngIf="addPaymentProviderForm.controls.username.hasError('required')">
+                Username is <strong>required</strong>
+              </mat-error>
+            </mat-form-field>
+          </div>
+          <div fxFlex="10%" fxLayout="column"></div>
+          <div fxFlex="40%" fxLayout="column">
+            <mat-form-field>
+              <mat-label>Password</mat-label>
+              <input matInput required formControlName="password" />
+              <mat-error *ngIf="addPaymentProviderForm.controls.password.hasError('required')">
+                Password is <strong>required</strong>
+              </mat-error>
+            </mat-form-field>
+          </div>
+        </div>
+      </mat-card-content>
+
+      <mat-card-actions fxLayout="row" fxLayout.xs="column" fxLayoutAlign="center" fxLayoutGap="5px">
+        <button type="button" mat-raised-button [routerLink]="['../']">Cancel</button>
+        <button
```

Comment:
@amydoxym for new submit buttons can we add the function Daniel added so there is a delay of 5 seconds before the button is enabled to avoid duplicates. @simiyu17 what is the name of that method and where can it be located.